File: src/IoTGame.Practice/Controllers/DataBaseController.cs
using System.Collections.Generic;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Data.Sqlite;

namespace IoTGame.Practice.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class DataController : ControllerBase
    {
        [HttpGet("query")]
        public IActionResult QueryData(string database, string table, string roadId)
        {
            string connectionString = GetConnectionString(database);
            var results = new List<object>();

            if (!System.IO.File.Exists(connectionString))
            {
                return Ok(results); // 返回空表
            }

            using (var connection = new SqliteConnection(connectionString))
            {
                connection.Open();

                var command = connection.CreateCommand();
                command.CommandText = $"SELECT date, [{roadId}] FROM {table}";
                command.Parameters.AddWithValue("@roadId", roadId);

                using (var reader = command.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        results.Add(new { Date = reader.GetString(0), Value = reader.GetInt32(1) });
                    }
                }
            }

            return Ok(results);
        }

        [HttpGet("history")]
        public IActionResult QueryHistoryData(string database, string table)
        {
            string connectionString = GetConnectionString(database);
            var results = new List<object>();

            if (!System.IO.File.Exists(connectionString))
            {
                return Ok(results); // 返回空表
            }

            using (var connection = new SqliteConnection(connectionString))
            {
                connection.Open();

                var command = connection.CreateCommand();
                command.CommandText = $"SELECT Id, Time, Flow FROM {table}";

                using (var reader = command.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        results.Add(
                            new
                            {
                                Id = reader.GetInt32(0),
                                Time = reader.GetString(1),
                                Flow = reader.GetFloat(2),
                            }
                        );
                    }
                }
            }

            return Ok(results);
        }

        private string GetConnectionString(string database)
        {
            return $"Data Source=../wwwroot/Data/{database}";
        }
    }
}

File: src/IoTGame.Practice/Controllers/DatabaseController.py
import sqlite3
import os

def get_connection_string(database):
    # 根据传入的数据库名称返回不同的连接字符串
    return f"../wwwroot/Data/{database}"

def test_query_data(database, road_id):
    connection_string = get_connection_string(database)
    results = []

    if not os.path.exists(connection_string):
        # 打印当前目录下的文件
        for file in os.listdir("../wwwroot/Data"):
            print(f"File: {file}")
        return results  # 返回空表

    try:
        conn = sqlite3.connect(connection_string)
        cursor = conn.cursor()
        cursor.execute(f"SELECT date, [{road_id}] FROM ExampleData")
        rows = cursor.fetchall()
        for row in rows:
            results.append({"Date": row[0], "Value": row[1]})
        # 打印部分读取的数据
        print(f"Sample data: {results[:5]}")
        conn.close()
    except sqlite3.Error as e:
        print(f"Database error: {e}")
        return []

    return results

# 测试函数调用
if __name__ == "__main__":
    database = "example_data.db"
    road_id = "1"
    data = test_query_data(database, road_id)
    print(data)
